invenio_cli/helpers/package_managers.py

```diff
@@ -171,3 +174,57 @@ def start_activated_subshell(self) -> List[str]:
         # Also, it has a good chance of not properly setting a PS1...
         shell = os.getenv("SHELL")
         return [shell, "-c", f"source .venv/bin/activate; exec {shell} -i"]
+
+
+class JavascriptPackageManager(ABC):
+    """Interface for creating tool-specific JS package management commands."""
+
+    name = None
+
+    def create_pynpm_package(self, package_json_path: Union[Path, str]) -> NPMPackage:
+        """Create a variant of ``NPMPackage`` with the path to ``package.json``."""
+        raise NotImplementedError()
+
+    def install_local_package(self, path: Union[Path, str]) -> List[str]:
+        """Install the local JS package."""
+        raise NotImplementedError()
+
+    def env_overrides(self) -> Dict[str, str]:
+        """Provide environment overrides for building Invenio assets."""
+        return {}
+
+
+class NPM(JavascriptPackageManager):
+    """Generate ``npm`` commands for managing JS packages."""
+
+    name = "npm"
+
+    def create_pynpm_package(self, package_json_path):
+        """Create an ``NPMPackage`` with the path to ``package.json``."""
+        return NPMPackage(package_json_path)
+
+    def install_local_package(self, path):
+        """Install the local JS package."""
+        return ["--prefix", str(path)]
+
+    def env_overrides(self):
+        """Provide environment overrides for building Invenio assets."""
+        return {"INVENIO_ASSETS_NPM_PKG_CLS": "pynpm:NPMPackage"}
```

Comment:
i think the name should not include NPM. i think something like `INVENIO_ASSETS_BUILDER_CLS` would be better

Reply:
sure, we can change the naming although "builder" would be a bit unfortunate since the builder would be webpack/rspack, not npm/pnpm which are the package managers

Reply:
LOL, you are right, `BUILDER` is not the right word for that. i saw `ASSETS` and thought about building the assets. maybe then `INVENIO_JAVASCRIPT_PACKAGE_MANAGER_CLS` :) i know it is long, but we are using internally also that word for npm/pnpm